### PR TITLE
Fix iceberg parquet table return empty result with date filter

### LINKED ISSUE #3004: Wrong result when query from iceberg external table with parquet format
### Steps to reproduce the behavior (Required)
1. create iceberg table and insert data:
```
create table hive_prod.iceberg_db.iceberg_parquet_snappy(
 col_boolean boolean,
 col_int int,
 col_long long,
 col_float float,
 col_double double,
 col_decimal decimal(38,18),
 col_date date,
 col_timestamp timestamp,
 col_string string,
 col_binary binary,
 col_struct struct<a:string,b:int>,
 col_map map<string,int>
)
USING iceberg 
TBLPROPERTIES('write.format.default'='parquet','write.parquet.compression-codec'='snappy');

insert into hive_prod.iceberg_db.iceberg_parquet_snappy select true,1,1.01,1.001,1.0001,10000000000.0000000001,cast('2020-01-01' as date),cast('2020-01-01 00:00:01' as timestamp),'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select false,2,2.02,2.002,2.0002,20000000000.0000000002,cast('2020-02-01' as date),cast('2020-02-01 00:00:02' as timestamp),'Xi Jinping, general secretary of the Communist Party of China Central Committee, called on Tuesday for unrelenting efforts in exercising full and rigorous governance over the Party, saying that the CPC will continue to show zero tolerance for corruption.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select true,3,3.03,3.003,3.0003,30000000000.0000000003,cast('2020-03-01' as date),cast('2020-03-01 00:00:04' as timestamp),'Nation raises caution on overseas packages',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select false,4,4.04,4.004,4.0004,40000000000.0000000004,cast('2020-04-01' as date),cast('2020-04-01 00:00:04' as timestamp),'Shanghai, Shenzhen register most newly listed firms in 2021',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select true,5,5.05,5.005,5.0005,50000000000.0000000005,cast('2020-05-01' as date),cast('2020-05-01 00:00:05' as timestamp),'A total of 4,685 companies have been listed on the A-share market in China as of Dec 31, 2021, with 46 percent of them based in Beijing, Shanghai, Shenzhen, Hangzhou, Suzhou, Guangzhou, Ningbo, Nanjing, Wuxi, and Chengdu, said a report of National Business Daily on Wednesday.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select false,6,6.06,6.006,6.0006,60000000000.0000000006,cast('2020-06-01' as date),cast('2020-06-01 00:00:06' as timestamp),'The development of enterprises, as well as the push of local governments, has boosted the listings, the report said. For example, in early 2018, Central China''s Hubei province launched a plan to double its listed companies by including the listing of enterprises into the government''s annual performance assessment.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select true,7,7.07,7.007,7.0007,70000000000.0000000007,cast('2020-07-01' as date),cast('2020-07-01 00:00:07' as timestamp),'Most listed companies in the top 10 cities come from industries such as information technology, electronics, mechanical equipment, medical biology, and electric power equipment, with information technology bearing most listed companies.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select false,8,8.08,8.008,8.0008,80000000000.0000000008,cast('2020-08-01' as date),cast('2020-08-01 00:00:08' as timestamp),'Citi report finds MNC mood in China buoyant',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select true,9,9.09,9.009,9.0009,90000000000.0000000009,cast('2020-09-01' as date),cast('2020-09-01 00:00:09' as timestamp),'Xi: China, Russia major champions of multilateralism, global justice',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select false,10,10.10,10.010,10.00010,100000000000.00000000010,cast('2020-10-01' as date),cast('2020-10-01 00:00:10' as timestamp),'Volkswagen deliveries hit by chip shortages',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select true,11,11.11,11.011,11.00011,110000000000.00000000011,cast('2020-11-01' as date),cast('2020-11-01 00:00:11' as timestamp),'Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select false,12,12.12,12.012,12.00012,120000000000.00000000012,cast('2020-12-01' as date),cast('2020-10-01 00:00:10' as timestamp),'The annual lantern show at Yuyuan Garden, a historic tourist site in Shanghai, will kick off on Tuesday and run through Feb 28.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_snappy select null,null,null,null,null,null,null,null,null,null,null,null;
```
 
2. create iceberg external table:
 ```
create external table ex_iceberg_tbl8 (
 col_boolean boolean,
 col_int int,
 col_long bigint,
 col_float float,
 col_double double,
 col_decimal decimal(38,18),
 col_date date,
 col_string varchar(200))
 ENGINE=iceberg
 properties (
 "resource" = "iceberg_emr_tn",
 "table" = "iceberg_parquet_snappy",
 "database" = "iceberg_db");
```

3. wrong result when query from iceberg external table:
```
mysql> select col_date from ex_iceberg_tbl8 where col_date >= '2020-01-01';
Empty set (0.09 sec)

mysql> select col_date from ex_iceberg_tbl8 where col_date between '2020-01-01' and '2020-06-01';
Empty set (0.09 sec)
```
col_date in iceberg external table:
```
mysql> select col_date from ex_iceberg_tbl8;
+------------+
| col_date   |
+------------+
| 2020-09-01 |
| 2020-12-01 |
| 2020-02-01 |
| 2020-10-01 |
| NULL       |
| 2020-11-01 |
| 2020-03-01 |
| 2020-08-01 |
| 2020-04-01 |
| 2020-07-01 |
| 2020-05-01 |
| 2020-01-01 |
| 2020-06-01 |
+------------+
13 rows in set (0.13 sec)
```
### Expected behavior (Required)
result from iceberg table:
```
spark-sql> select col_date from hive_prod.iceberg_db.iceberg_orc_snappy where col_date >= '2020-01-01';
2020-10-01
2020-06-01
2020-08-01
2020-09-01
2020-03-01
2020-01-01
2020-12-01
2020-11-01
2020-07-01
2020-04-01
2020-02-01
2020-05-01
Time taken: 0.757 seconds, Fetched 12 row(s)
spark-sql>  select col_date from hive_prod.iceberg_db.iceberg_orc_snappy where col_date between '2020-01-01' and '2020-06-01';
2020-06-01
2020-03-01
2020-01-01
2020-04-01
2020-02-01
2020-05-01
Time taken: 0.481 seconds, Fetched 6 row(s)
```

### Real behavior (Required)
no result returned

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER d404b84 |
+------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [x] bug

## Which issues of this PR fixes ：

Fixes #
close #3004 

## Problem Summary(Required) ：

Expression converter will get incorrect epoch day for Date type which will cause query with invalid result.
DateTime will be support in an other PR.
```
MySQL [external_db]>  select col_date from ex_iceberg_tbl8 where col_date >= '2020-01-01';
Empty set (2.79 sec)
```
after this pr
```
MySQL [external_db]>  select col_date from ex_iceberg_tbl8 where col_date >= '2020-01-01';
+------------+
| col_date   |
+------------+
| 2020-02-01 |
| 2020-04-01 |
| 2020-08-01 |
| 2020-06-01 |
| 2020-12-01 |
| 2020-01-01 |
| 2020-03-01 |
| 2020-09-01 |
| 2020-07-01 |
| 2020-05-01 |
| 2020-10-01 |
| 2020-11-01 |
+------------+
```

### CODE REVIEW

**Comment by caneGuy** [comment]
i will fix the ut later. could you help review this ? @DorianZheng @Ielihs thanks

**Comment by DorianZheng** [comment]
Plz add UT. rest LGTM

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Comment by caneGuy** [comment]
run starrocks_fe_unittest

**Comment by DorianZheng** [comment]
https://github.com/StarRocks/starrocks/pull/2792 will optimize it.

**Comment by caneGuy** [comment]
UT has run successfully @DorianZheng PTAL thanks

**Review by Youngwb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/ExpressionConverter.java [hunk 1]
```diff
@@ -84,13 +84,15 @@ public static UnboundPredicate toIcebergExpression(Expr expr) {
...
+                        Math.toIntExact(((DateLiteral) literal).toLocalDateTime()
+                                .toLocalDate().toEpochDay()));
                 break;
             }
-            default: break;
+            case DATETIME:
+            default:
+                break;
         }
         if (unboundPredicate == null) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/ExpressionConverter.java:97]: This check is useless and can be removed.

Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/ExpressionConverter.java]: thanks @wyb PTAL

**Review by wyb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**